### PR TITLE
Source kit for React

### PR DESCRIPTION
**Pull request by kellyeryan**
#### Screens

Screen Shot 2020-05-23 at 11 41 43 AM

Screen Shot 2020-05-23 at 11 41 50 AM

Screen Shot 2020-05-23 at 11 41 59 AM

Screen Shot 2020-05-23 at 11 42 06 AM

#### Breaking Changes

None

#### Github Issue

#227

#### Checklist:

- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review
- [X] **SCREENSHOT** Please add a screen shot or two
- [x] **SPECS** Please cover your changes with specs - NA - React only
- [x] **CHANGELOG** Please add an entry on `[Unreleased]` section to every functionality `ADDED`/`CHANGED`/`DEPRECATED`/`REMOVED`/`FIXED`

### CODE REVIEW

File: app/pb_kits/playbook/pb_source/_source.jsx [hunk 1]
```diff
@@ -1,21 +1,129 @@
...
+  const dataProps = buildDataProps(data)
+
+  const css = classnames([
+    'pb_source_kit',
+    className,
+  ])
+
+  const avatar = () => {
+    if ((type === 'user' || type === 'referral') && user.name !== undefined) {
+      const avatarProps = Object.assign({}, user)
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_source/_source.jsx:48]: One this part, I'm wondering about the need for `Object.assign...`. I'm unsure if you need it as the default on `:37` should be able to cover this. If you are just cloning here, you will want to go with:

```js
const avatarProps = { ...user }
```

[reference](https://thecodebarbarian.com/object-assign-vs-object-spread.html)

File: app/pb_kits/playbook/pb_source/_source.jsx [hunk 2]
```diff
@@ -1,21 +1,129 @@
...
+      avatarProps['size'] = 'sm'
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_source/_source.jsx:49]:
```suggestion
      avatarProps.size = 'sm'
```

File: app/pb_kits/playbook/pb_source/_source.jsx [hunk 3]
```diff
@@ -1,21 +1,129 @@
...
+    case 'prospecting':
+      return 'binoculars'
+    case 'referral':
+      return 'handshake'
+    case 'retail':
+      return 'shopping-bag'
+    case 'inbound':
+      return 'sign-in'
+    }
+  }
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_source/_source.jsx:78]:
```suggestion
  const typeIconNames = {
    events: 'calendar-alt',
    outbound: 'sign-out',
    prospecting: 'binoculars',
    referral: 'handshake',
    retail: 'shopping-bag',
    inbound: 'sign-in',
  }
```

File: app/pb_kits/playbook/pb_source/_source.jsx [hunk 4]
```diff
@@ -1,21 +1,129 @@
...
+        {...dataProps}
+        className={css}
+        id={id}
+    >
+
+      <div className="pb__source_layout">
+        <If condition={hideIcon === false}>
+          <If condition={showIcon()}>
+            <IconCircle
+                icon={typeIconName()}
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_source/_source.jsx:93]:
```suggestion
                icon={typeIconNames[type]}
```

File: app/pb_kits/playbook/pb_source/_source.jsx (hunk 3 above)
Comment (thestephenmarshall) [reply app/pb_kits/playbook/pb_source/_source.jsx:78]: Perhaps we could also move this out of the closure as well.

**Comment by kellyeryan** [comment]
Thanks so much for the comments and feedback! I've made the changes, double-checked I didn't break anything, and pushed it up again.

**Pull request merged**